app/controllers/identity.js

```diff
@@ -14,16 +14,40 @@ export default class IdentityController extends Controller {
   @tracked isCopyClicked = false;
   @tracked isVerifyClicked = false;
   @tracked profileURL = this.model.profileURL || '';
+  @tracked saveDisabled = true;
+  @tracked generateChainCodeDisabled = this.model.profileURL === undefined;
+  @tracked checkboxDisabled =
+    this.generateChainCodeDisabled || this.model.chaincode === undefined;
 
   @action async handleRefresh() {
     window.location.reload();
   }
 
+  @action changeSaveDisabled() {
+    const isValidUrl = (urlString) => {
+      try {
+        return Boolean(new URL(urlString));
+      } catch (e) {
+        return false;
+      }
+    };
+    if (
+      this.profileURL === '' ||
+      this.profileURL === (this.model.profileURL || '') ||
+      !isValidUrl(this.profileURL)
```

Comment:
Instead of checking this much conditions, is it possible to check for when the `saveDisabled` is set to true ??
In short can we simplify this ??

Reply:
These are necessary conditions to check, therefore may not be simplified